### PR TITLE
Auto Migration Tests (Part 1 - auto migration mysql types, default size setting)

### PR DESCRIPTION
**Pull request by TimothyGillespie**
Only experementing until now. I have a different idea how I wanna design these to make it more scalable and easier to maintain.

### CODE REVIEW

File: src/test/java/org/javawebstack/orm/test/DatabaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.javawebstack.orm.wrapper.MySQL;
+import org.javawebstack.orm.wrapper.SQL;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public abstract class DatabaseConnection {
```
Comment (JWaldecker) [inline src/test/java/org/javawebstack/orm/test/DatabaseConnection.java]: I'm not sure if an abstract class is the best way to to this, e.g. a user may use connections to multiple databases. What do you think about it?

Comment (TimothyGillespie) [reply src/test/java/org/javawebstack/orm/test/DatabaseConnection.java]: This is only for the test so the user won't really see, use or be affected byit.

Comment (JWaldecker) [reply src/test/java/org/javawebstack/orm/test/DatabaseConnection.java]: Okay, but with this we are limited to this one extends from ORMTestCase

Comment (TimothyGillespie) [reply src/test/java/org/javawebstack/orm/test/DatabaseConnection.java:11]: Can you come to teamspeak to talk about this?

Comment (JanHolger) [reply src/test/java/org/javawebstack/orm/test/DatabaseConnection.java]: He moved the sql connection logic out of the base test class (ORMTestCase) and ORMTestCase aswell as any potential future base class can extend DatabaseConnection aswell to use the MySQL Connection. A single database should be totally fine for our testing purposes. Everything in src/test is compiled but not packaged into the library, so you won't be able to use this base class in your application's tests anyway. We could introduce something like that but in an application environment you are most likely going to fire up your entire application anyway so it's logic will handle connections.

**Comment by TimothyGillespie** [comment]
To prevent this from becoming a gigantic PR, which nobody wants to review I started to section this off.

**Review by JanHolger — APPROVED** [review]
I would maybe rename the `testenv.properties` to just `test.properties` but overall it looks ok

**Pull request merged**